apps/api/src/domain/games/memory-match/services/memory-match.service.ts

```diff
@@ -157,4 +241,99 @@ export class MemoryMatchService {
             'NODE_PORT'
         )}/${this.configService.get('GLOBAL_PREFIX')}/resources/mediaItems/download/${mediaItemId}`;
     }
+
+    private async validateMediaItemReferences(
+        memoryMatchRound: MemoryMatchRound
+    ): Promise<InternalError[]> {
+        if (!isNonEmptyObject(memoryMatchRound)) {
+            return [];
+        }
+
+        const { cardBackImageId } = memoryMatchRound;
+
+        const imageIds = memoryMatchRound.cards.map(({ imageId }) => imageId);
+
+        if (!isNullOrUndefined(cardBackImageId)) {
+            imageIds.push(cardBackImageId);
+        }
+
+        const audioIds = memoryMatchRound.cards.map(({ audioId }) => audioId);
+
+        const mediaItemIds = [...imageIds, ...audioIds];
+
+        /**
+         * TODO Optimize this to use a filter in-database
```

Comment:
We should do this now because we already have a large number of media items in the DB.

Reply:
[Here](https://coscrad.atlassian.net/browse/CWEBJIRA-325) is the story. We'll do this as soon as I complete the new user-defined filters support in Arango, which we can immediately leverage to filter by ID and MIMEType.